### PR TITLE
Issue 384 wdl

### LINKED ISSUE #384: Acquire WDL for posting to GATK protected, if possible
Need to get from @dlivitz 

Put into ``scripts/wdl/``

Should coincide with #383

### PR DESCRIPTION
**Pull request by LeeTL1220**
Closes #384

### CODE REVIEW

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 1]
```diff
@@ -0,0 +1,304 @@
+#
+# Workflow a single pair of case-control samples.
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Workflow *for* . . . ?

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+        entity_id=wf_entity_id_tumor,
+        padded_target_bed=PadTargets.padded_target_bed,
+        input_bam=tumor_bam,
+        ref_fasta=ref_fasta,
+        ref_fasta_fai=ref_fasta_fai,
+        ref_fasta_dict=ref_fasta_dict,
+        jar_file=jar_file,
+        disable_reference_validation=is_disable_reference_validation,
+        mem=2
+
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: extra whitespace line

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+  call CalculateTargetCoverage as TumorCalculateTargetCoverage {
+    input:
+        entity_id=wf_entity_id_tumor,
+        padded_target_bed=PadTargets.padded_target_bed,
+        input_bam=tumor_bam,
+        ref_fasta=ref_fasta,
+        ref_fasta_fai=ref_fasta_fai,
+        ref_fasta_dict=ref_fasta_dict,
+        jar_file=jar_file,
+        disable_reference_validation=is_disable_reference_validation,
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: why the different names (i.e. `is_`) on the inner scope vs outer scope variable names?

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+
+  call NormalizeSomaticReadCounts as TumorNormalizeSomaticReadCounts {
+    input:
+        entity_id=wf_entity_id_tumor,
+        coverage_file=TumorCalculateTargetCoverage.gatk_cnv_coverage_file,
+        padded_target_bed=PadTargets.padded_target_bed,
+        pon=PoN,
+        jar_file=jar_file,
+        mem=2,
+	jni_lib=jni_lib
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: indentation

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+        entity_id=wf_entity_id_normal,
+        padded_target_bed=PadTargets.padded_target_bed,
+        input_bam=tumor_bam,
+        ref_fasta=ref_fasta,
+        ref_fasta_fai=ref_fasta_fai,
+        ref_fasta_dict=ref_fasta_dict,
+        jar_file=jar_file,
+        disable_reference_validation=is_disable_reference_validation,
+        mem=2
+
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: extra whitespace line

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+  call NormalizeSomaticReadCounts as NormalNormalizeSomaticReadCounts {
+    input:
+        entity_id=wf_entity_id_normal,
+        coverage_file=NormalCalculateTargetCoverage.gatk_cnv_coverage_file,
+        padded_target_bed=PadTargets.padded_target_bed,
+        pon=PoN,
+        jar_file=jar_file,
+        mem=2,
+	jni_lib=jni_lib
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: indentation

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+
+  call PerformSegmentation as NormalPerformSeg {
+    input:
+        entity_id=wf_entity_id_normal,
+        jar_file=jar_file,
+        tn_file=NormalNormalizeSomaticReadCounts.tn_file,
+        mem=2
+  }
+
+  call Caller as NormalCaller {
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Just checking that you really intend to segment the normal sample.

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+        seg_file=NormalPerformSeg.seg_file,
+        mem=2
+  }
+
+  call HetPulldown {
+    input:
+        entity_id_tumor=wf_entity_id_tumor,
+        entity_id_normal=wf_entity_id_normal,
+        jar_file=jar_file,
+        mem = 4,
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: For consistency, you should put `mem` at the bottom.

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+        normal_bam=normal_bam,
+        normal_bam_idx=normal_bam_idx,
+        common_snp_list=common_snp_list
+  }
+
+  call AllelicCNV {
+    input:
+        entity_id=wf_entity_id_tumor,
+        jar_file=jar_file,
+        mem=4,
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Ditto, `mem` at the bottom

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+        tumor_hets=HetPulldown.tumor_hets,
+        called_file=TumorCaller.called_file,
+        tn_file=TumorNormalizeSomaticReadCounts.tn_file
+  }
+}
+
+# Pad the target file.  This was found to help sensitivity and specificity.  This step should only be altered
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Single-space indent followed by double-space indent on next line.

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+#  by advanced users.  Note that by changing this, you need to have a PoN that also reflects the change.
+task PadTargets {
+    File target_bed
+    Int pd = 250
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: You can name the variable `padding` without running afoul of the verbosity police.

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 12]
```diff
@@ -0,0 +1,304 @@
...
+    }
+    output {
+        File padded_target_bed = "targets.padded.bed"
+    }
+    #runtime {
+    #    docker: "gatk-protected/a1"
+    #}
+}
+
+# Calculate the target coverage
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Comment should note that this is pcov.

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 13]
```diff
@@ -0,0 +1,304 @@
...
+task CalculateTargetCoverage {
+    String entity_id
+    File padded_target_bed
+    String transform = "PCOV"
+    String grouping = "SAMPLE"
+    Boolean keepduplicatereads = true
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: underscores in variable name

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 14]
```diff
@@ -0,0 +1,304 @@
...
+
+# Pad the target file.  This was found to help sensitivity and specificity.  This step should only be altered
+#  by advanced users.  Note that by changing this, you need to have a PoN that also reflects the change.
+task PadTargets {
+    File target_bed
+    Int pd = 250
+    Int mem
+    File jar_file
+    command {
+        java -Xmx${mem}g -jar ${jar_file} PadTargets  --targets ${target_bed} --output targets.padded.bed --padding ${pd}  --help false --version false --verbosity INFO --QUIET false
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Break up command line with a `\` separator.

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 15]
```diff
@@ -0,0 +1,304 @@
...
+    File input_bam
+    File ref_fasta
+    File ref_fasta_fai
+    File ref_fasta_dict
+    Int mem
+    File jar_file
+    String disable_reference_validation
+
+    command {
+        java -Xmx${mem}g -jar ${jar_file} CalculateTargetCoverage --output ${entity_id}.coverage.tsv --groupBy ${grouping} --transform ${transform} --targets ${padded_target_bed} --targetInformationColumns FULL --keepduplicatereads ${keepduplicatereads} --input ${input_bam} --reference ${ref_fasta}  --disable_all_read_filters ${disable_all_read_filters} --interval_set_rule UNION --interval_padding 0 --secondsBetweenProgressUpdates 10.0 --disableSequenceDictionaryValidation ${disable_reference_validation} --createOutputBamIndex true --help false --version false --verbosity INFO --QUIET false
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Likewise, split the long line.

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 16]
```diff
@@ -0,0 +1,304 @@
...
+    File pon
+    Int mem
+    File jar_file
+    File jni_lib
+    command {
+        java -Xmx${mem}g -Djava.library.path=${jni_lib} -jar ${jar_file} NormalizeSomaticReadCounts  --input ${coverage_file} \
+        --targets ${padded_target_bed} --panelOfNormals ${pon} --factorNormalizedOutput ${entity_id}.fnt.tsv --tangentNormalized ${entity_id}.tn.tsv \
+        --betaHatsOutput ${entity_id}.betaHats.tsv --preTangentNormalized  ${entity_id}.preTN.tsv  --help false --version false --verbosity INFO --QUIET false
+    }
+
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: extra whitespace line

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 17]
```diff
@@ -0,0 +1,304 @@
...
+    String entity_id
+    File coverage_file
+    File padded_target_bed
+    File pon
+    Int mem
+    File jar_file
+    File jni_lib
+    command {
+        java -Xmx${mem}g -Djava.library.path=${jni_lib} -jar ${jar_file} NormalizeSomaticReadCounts  --input ${coverage_file} \
+        --targets ${padded_target_bed} --panelOfNormals ${pon} --factorNormalizedOutput ${entity_id}.fnt.tsv --tangentNormalized ${entity_id}.tn.tsv \
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Indent this and the next continued line.

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 18]
```diff
@@ -0,0 +1,304 @@
...
+    File jar_file
+    File tn_file
+
+    command {
+        java -Xmx${mem}g -jar ${jar_file} PerformSegmentation  --targets ${tn_file} \
+        --output ${entity_id}.seg --log2Input true  --alpha 0.01 --nperm 10000 --pmethod HYBRID --minWidth 2 --kmax 25 \
+        --nmin 200 --eta 0.05 --trim 0.025 --undoSplits NONE --undoPrune 0.05 --undoSD 3 --help false --version false \
+        --verbosity INFO --QUIET false
+    }
+
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: extra whitespace line

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 19]
```diff
@@ -0,0 +1,304 @@
...
+    output {
+        File gatk_cnv_coverage_file = "${entity_id}.coverage.tsv"
+    }
+
+    #runtime {
+    #    docker: "gatk-protected/a1"
+    #}
+}
+
+task NormalizeSomaticReadCounts {
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Needs comment/doc

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 20]
```diff
@@ -0,0 +1,304 @@
...
+        File tn_file = "${entity_id}.tn.tsv"
+        File pre_tn_file = "${entity_id}.preTN.tsv"
+        File betahats_file = "${entity_id}.betaHats.tsv"
+    }
+    #runtime {
+    #    docker: "gatk-protected/a1"
+    #}
+}
+
+task PerformSegmentation {
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Needs comment/doc.  It should note something about the unexposed (to the WDL script, that is) CBS parameters.

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 21]
```diff
@@ -0,0 +1,304 @@
...
+        File called_file="${entity_id}.called"
+    }
+}
+
+# entity IDs can be the same value
+task HetPulldown {
+    String entity_id_tumor
+    String entity_id_normal
+    File jar_file
+    Int mem
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: `mem` at the bottom for consistency

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 22]
```diff
@@ -0,0 +1,304 @@
...
+    File normal_bam_idx
+    File common_snp_list
+
+    command {
+        java -Xmx${mem}g -jar ${jar_file} GetHetCoverage  --reference ${ref_fasta} \
+        --normal ${normal_bam} --tumor ${tumor_bam} --snpIntervals ${common_snp_list}  \
+        --normalHets ${entity_id_normal}.normal.hets.tsv --tumorHets ${entity_id_tumor}.tumor.hets.tsv --pvalueThreshold 0.05 \
+        --help false --version false --verbosity INFO --QUIET false --VALIDATION_STRINGENCY LENIENT
+    }
+
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: extra whitespace line

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 23]
```diff
@@ -0,0 +1,304 @@
...
+    File ref_fasta_dict
+    File tumor_bam
+    File tumor_bam_idx
+    File normal_bam
+    File normal_bam_idx
+    File common_snp_list
+
+    command {
+        java -Xmx${mem}g -jar ${jar_file} GetHetCoverage  --reference ${ref_fasta} \
+        --normal ${normal_bam} --tumor ${tumor_bam} --snpIntervals ${common_snp_list}  \
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: indent the continued lines

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 24]
```diff
@@ -0,0 +1,304 @@
...
+         --segments ${seg_file} --output ${entity_id}.called --threshold 2.0  --legacy false --experimental false \
+          --help false --version false --verbosity INFO --QUIET false
+    }
+
+    output {
+        File called_file="${entity_id}.called"
+    }
+}
+
+# entity IDs can be the same value
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Needs doc

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 25]
```diff
@@ -0,0 +1,304 @@
...
+        --help false --version false --verbosity INFO --QUIET false --VALIDATION_STRINGENCY LENIENT
+    }
+
+    output {
+        File normal_hets="${entity_id_normal}.normal.hets.tsv"
+        File tumor_hets="${entity_id_tumor}.tumor.hets.tsv"
+    }
+}
+
+task AllelicCNV {
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Needs doc

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 26]
```diff
@@ -0,0 +1,304 @@
...
+    String entity_id
+    File jar_file
+    Int mem
+    File tumor_hets
+    File called_file
+    File tn_file
+
+    command {
+        java -Xmx${mem}g -jar ${jar_file} AllelicCNV  --tumorHets ${tumor_hets} \
+         --tangentNormalized ${tn_file} --segments ${called_file} --outputPrefix ${entity_id} \
```
Comment (davidbenjamin) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: indent continued lines

**Comment by davidbenjamin** [comment]
Review complete.

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 1 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 3 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 2 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: So bad you had to say it twice....

Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 4 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 7 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Yes, needed for @kareembroad tool.

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 8 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 9 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 10 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: I do that to indicate the continuation of a sentence.

No action

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 11 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 14 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 12 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 13 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 15 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 19 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 17 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 16 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Keeping that whitespace in all tasks.

Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 20 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Filed an issue that we should expose the parameters.

Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 18 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Keeping these whitespaces and corrected above one to add a whitespace.

No action.

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 24 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 21 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done here and elsewhere.

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 23 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 22 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: See above

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 25 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 26 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Done

**Comment by LeeTL1220** [comment]
@davidbenjamin Addressed comments and added template json file.  Please take a look at my responses and that json file.

**Comment by davidbenjamin** [comment]
@LeeTL1220 I'm satisfied and will rebase and merge.

**Pull request merged**